lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -180,6 +180,10 @@ def load_resource(klass:, search:, permission:, batch: 1_000, includes: nil, lim
             base = base.select(select) unless select.nil?
             base.in_batches(of: batch)
           end
+
+          def version(version)
+            Gem::Version.new(version.to_s)
```

Comment:
potentially we can rescue from ArgumentError which is raised on malformed string and return Gem::Version.new(0) (nil is discouraged), though that can be misleading in comparisons, e.g. it will always be smaller than any other version

perhaps raising some better rendering error would be best, thoughts?

Reply:
How difficult would using a proxy object (That we control) be?

Reply:
Not something I'd do in next month :-) do you have some more complicated use of it in mind? Or you don't want to rely on Gem not chaning it's API?

Reply:
The latter. I fear we‘re introducing technical depth here.

Reply:
You could do what Puppet has done: a [versioncmp](https://puppet.com/docs/puppet/latest/function.html#versioncmp) function that is basically `Gem::Version.new(first) <=> Gem::Version.new(second)`. That API might be easier to guarantee in the future.

For reference, I know that RPM and Debian version comparisons are slightly more complex and different than semver. Probably good to at least state what it's based on.

Reply:
What Ewoud says comes to my mind too, the most clean way. Hides the implementation completely.